runelite-client/src/main/java/net/runelite/client/plugins/microbot/GirdyScripts/cannonballsmelter/CannonballSmelterScript.java

```diff
@@ -0,0 +1,219 @@
+package net.runelite.client.plugins.microbot.GirdyScripts.cannonballsmelter;
+
+
+import net.runelite.api.Client;
+import net.runelite.api.TileObject;
+import net.runelite.client.plugins.microbot.Microbot;
+import net.runelite.client.plugins.microbot.Script;
+import net.runelite.client.plugins.microbot.GirdyScripts.cannonballsmelter.enums.CannonballSmelterStates;
+import net.runelite.client.plugins.microbot.util.antiban.Rs2Antiban;
+import net.runelite.client.plugins.microbot.util.antiban.Rs2AntibanSettings;
+import net.runelite.client.plugins.microbot.util.antiban.enums.Activity;
+import net.runelite.client.plugins.microbot.util.bank.Rs2Bank;
+import net.runelite.client.plugins.microbot.util.gameobject.Rs2GameObject;
+import net.runelite.client.plugins.microbot.util.inventory.Rs2Inventory;
+import net.runelite.client.plugins.microbot.util.keyboard.Rs2Keyboard;
+import net.runelite.client.plugins.microbot.util.player.Rs2Player;
+import net.runelite.client.plugins.microbot.util.widget.Rs2Widget;
+
+import javax.inject.Inject;
+import java.awt.event.KeyEvent;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+
+public class CannonballSmelterScript extends Script {
+
+    public static String version = "1.0.0";
+    private final ThreadLocalRandom random = ThreadLocalRandom.current();
+    @Inject
+    private CannonballSmelterConfig config;
+    @Inject
+    private Client client;
+
+    long startTime;
+    long endTime;
+
+    CannonballSmelterStates state = CannonballSmelterStates.IDLING;
+
+
+    private boolean hasBalls() {
+        return Rs2Inventory.hasItem(2);
+    }
+    private boolean hasBars() {
+        return Rs2Inventory.hasItem(2353);
+    }
+    private boolean required() {return Rs2Inventory.hasItem("ammo mould");}
+
+    public boolean run(CannonballSmelterConfig config) {
+        Rs2Antiban.resetAntibanSettings();
+        cannonballAntiBan();
+        Rs2AntibanSettings.actionCooldownChance = 0.1;
+        mainScheduledFuture = scheduledExecutorService.scheduleWithFixedDelay(() -> {
```

Comment:
You can add 

Microbot.enableAutoRunOn = true;
Microbot.runEnergyThreshold = 5000; // 50 run energy

The bot will automatically start running when you have 50 or more run energy and will use any stamina/run energy potions if they are in the inventory.

This way you don't have to handle toggleRunEnergy on script level.